language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/DamlGenMap.java

```diff
@@ -22,15 +21,30 @@ public DamlGenMap(java.util.Map<Value, Value> value) {
         this.value = value;
     }
 
+    public static <T> Collector<T, Map<Value, Value>, DamlGenMap> collector(
+            Function<T, Value> keyMapper,
+            Function<T, Value> valueMapper) {
+
+        return Collector.of(
+                LinkedHashMap::new,
+                (acc, entry) -> acc.put(keyMapper.apply(entry), valueMapper.apply(entry)),
+                (left, right) -> {
+                    left.putAll(right);
+                    return left;
+                },
+                DamlGenMap::new
+        );
+    }
+
     public @Nonnull
     java.util.Map<Value, Value> getMap() { return value; }
```

Comment:
If the map is supposed to be immutable, might wanna return `Collections.unmodifiableMap(value)` here.

Reply:
In fact if we do not copy the map given as argument in the constructor, this is pretty useless.